sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeBatchActionsAsyncClient.java

```diff
@@ -171,7 +173,9 @@ private JobManifestTasks getJobManifestTasks(TextAnalyticsActions actions) {
                             // temporally set the default value to 'latest' until service correct it.
                             // https://github.com/Azure/azure-sdk-for-java/issues/17625
                             new EntitiesTaskParameters()
-                                .setModelVersion(getNotNullModelVersion(action.getModelVersion())));
+                                .setModelVersion(getNotNullModelVersion(action.getModelVersion()))
+                                .setStringIndexType(StringIndexTypeResponse.fromString(
+                                    getNonNullStringIndexType(action.getStringIndexType()).toString())));
```

Comment:
Should we make this `getNonNullStringIndexType` return StringIndexTypeResponse.fromString( directly from the method, that way we can reduce the duplicate lines.